cmd/ui/src/views/Explore/BasicObjectInfoFields.tsx

```diff
@@ -0,0 +1,93 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+import { Box } from "@mui/material";
+import { NodeIcon, Field } from "bh-shared-ui";
+import { TIER_ZERO_TAG } from "src/constants";
+import { GraphNodeTypes } from "src/ducks/graph/types";
+import { setSearchValue, startSearchSelected } from "src/ducks/searchbar/actions";
+import { PRIMARY_SEARCH, SEARCH_TYPE_EXACT } from "src/ducks/searchbar/types";
+import { useAppDispatch } from "src/store";
+
+interface BasicObjectInfoFieldsProps {
+    objectid: string;
+    displayname?: string;
+    system_tags?: string;
+    service_principal_id?: string;
+    noderesourcegroupid?: string;
+    name?: string;
+}
+
+const RelatedKindField = (fieldLabel: string, relatedKind: GraphNodeTypes, id: string, name?: string) => {
+    const dispatch = useAppDispatch();
+    return (
+        <Box padding={1}>
+            <Box fontWeight='bold' mr={1}>
+                {fieldLabel}
+            </Box>
+            <br />
+            <Box display='flex' flexDirection='row' flexWrap='wrap' justifyContent='flex-start'>
+                <NodeIcon nodeType={relatedKind} />
+                <Box
+                    onClick={() => {
+                        dispatch(
+                            setSearchValue(
+                                {
+                                    objectid: id,
+                                    label: '',
+                                    type: relatedKind,
+                                    name: name || '',
+                                },
+                                PRIMARY_SEARCH,
+                                SEARCH_TYPE_EXACT
+                            )
+                        );
+                        dispatch(startSearchSelected(PRIMARY_SEARCH));
+                    }}
+                    style={{ cursor: 'pointer' }}
+                    overflow='hidden'
+                    textOverflow='ellipsis'
+                    title={id}>
+                    {id}
+                </Box>
+            </Box>
+        </Box>
+    );
+};
+
+export const BasicObjectInfoFields: React.FC<BasicObjectInfoFieldsProps> = (props): JSX.Element => {
+    return (
+        <>
+            {props.system_tags?.includes(TIER_ZERO_TAG) && <Field label='Tier Zero:' value={true} />}
+            {props.displayname && <Field label='Display Name:' value={props.displayname} />}
+            <Field label='Object ID:' value={props.objectid} />
+            {props.service_principal_id &&
+                RelatedKindField(
+                    'Service Principal ID:',
+                    GraphNodeTypes.AZServicePrincipal,
+                    props.service_principal_id,
+                    props.name
+                )}
```

Comment:
oops I totally see now that these changes are from migrating code into `bh-share-ui` - totally feel free to disregard the above comment lol!

Reply:
I'm pretty sure I wrote this. If it's not a big lift it would be nice to clean this up but as you mention, Ben, Wes is only reorganizing the existing code.